### PR TITLE
SONARGRADL-92 Ignore configuration cache in plugin

### PR DESCRIPTION
**Pull request by leo-geoffroy-sonarsource**

### CODE REVIEW

File: integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java [hunk 1]
```diff
@@ -227,4 +226,24 @@ public void testJaCoCoProperties() throws Exception {
     assertThat(props.getProperty("sonar.jacoco.reportPaths")).contains(baseDir.resolve("build/jacoco/test.exec").toString());
     assertThat(props.getProperty("sonar.coverage.jacoco.xmlReportPaths")).contains(baseDir.resolve("build/reports/jacoco/test/jacocoTestReport.xml").toString());
   }
+
+  @Test
+  public void testProjectWithConfigurationCacheDoubleExecutionsShouldWork() throws Exception {
+    //--configuration cache was brought by release 6.5.0
+    if (getGradleVersion().isGreaterThan("6.5.0")) {
```
Comment (dbmeneses) [inline integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java:233]: wouldn't it be better to use junit's `Assume.assumeTrue`? That way it shows as skipped.

File: integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java [hunk 2]
```diff
@@ -227,4 +226,24 @@ public void testJaCoCoProperties() throws Exception {
...
+      runGradlewSonarQubeWithEnv("/java-gradle-configuration-cache", emptyMap(), "-Dsonar.scanner.dumpToFile=asd", "--configuration-cache");
+      RunResult runResult = runGradlewSonarQubeWithEnv("/java-gradle-configuration-cache", emptyMap(), "-Dsonar.scanner.dumpToFile=asd", "--configuration-cache");
+
+      //Versions lower than 7.4.0 doesn't support org.gradle.api.internal.AbstractTask#notCompatibleWithConfigurationCache, so it is not possible to prevent configuration cache
+      if (getGradleVersion().isLowerThan("7.4.0")) {
+        assertThat(runResult.getLog()).contains("no properties configured, was it skipped in all projects?");
+        assertThat(runResult.getLog()).contains("BUILD SUCCESSFUL");
+      }else{
```
Comment (dbmeneses) [inline integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java:241]: formatting not right?

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java [hunk 3]
```diff
@@ -99,6 +100,17 @@ private void configureTask(SonarQubeTask sonarQubeTask, Project project, Map<Str
...
+  }
+
+  private static void setNotCompatibleWithConfigurationCache(SonarQubeTask sonarQubeTask) {
+    if (isGradleVersionGreaterThan("7.4.0")) {
+      sonarQubeTask.notCompatibleWithConfigurationCache("Plugin is not compatible with configuration cache");
+    }
+  }
+
+  private static boolean isGradleVersionGreaterThan(String version) {
+    return GradleVersion.current().compareTo(GradleVersion.version(version)) >=0;
```
Comment (dbmeneses) [inline src/main/java/org/sonarqube/gradle/SonarQubePlugin.java:113]: Looks like the comparison is "greater or equals" which doesn't match the method name?

**Review by dbmeneses — APPROVED** [review]

File: integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java [hunk 4]
```diff
@@ -227,4 +226,24 @@ public void testJaCoCoProperties() throws Exception {
     assertThat(props.getProperty("sonar.jacoco.reportPaths")).contains(baseDir.resolve("build/jacoco/test.exec").toString());
     assertThat(props.getProperty("sonar.coverage.jacoco.xmlReportPaths")).contains(baseDir.resolve("build/reports/jacoco/test/jacocoTestReport.xml").toString());
   }
+
+  @Test
+  public void testProjectWithConfigurationCacheDoubleExecutionsShouldWork() throws Exception {
+    //--configuration cache was brought by release 6.5.0
+    if (getGradleVersion().isGreaterThan("6.5.0")) {
+      runGradlewSonarQubeWithEnv("/java-gradle-configuration-cache", emptyMap(), "-Dsonar.scanner.dumpToFile=asd", "--configuration-cache");
+      RunResult runResult = runGradlewSonarQubeWithEnv("/java-gradle-configuration-cache", emptyMap(), "-Dsonar.scanner.dumpToFile=asd", "--configuration-cache");
```
Comment (dbmeneses) [inline integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java:235]: Did we need a dedicated project for this test?

File: src/main/java/org/sonarqube/gradle/SonarQubePlugin.java (hunk 3 above)
Comment (leo-geoffroy-sonarsource) [reply src/main/java/org/sonarqube/gradle/SonarQubePlugin.java:113]: Done

File: integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java (hunk 2 above)
Comment (leo-geoffroy-sonarsource) [reply integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java:241]: Formatted

File: integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java (hunk 1 above)
Comment (leo-geoffroy-sonarsource) [reply integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java:233]: Didn't know about that, that's a good one :)

File: integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java (hunk 4 above)
Comment (leo-geoffroy-sonarsource) [reply integrationTests/src/test/java/org/sonarqube/gradle/GradleTest.java]: Thank you for the catch, there is no need for a dedicated project in that case

**Pull request merged**